src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/Rfc2898DeriveBytes.OneShot.cs

```diff
@@ -0,0 +1,128 @@
+        }
+
+        public static byte[] Pbkdf2DeriveBytes(
+            ReadOnlySpan<byte> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            int length,
+            HashAlgorithmName hashAlgorithm)
+        {
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            byte[] result = new byte[length];
+            Pbkdf2DeriveBytes(password, salt, iterations, hashAlgorithm, result);
+            return result;
+        }
+
+        public static void Pbkdf2DeriveBytes(
+            ReadOnlySpan<byte> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            HashAlgorithmName hashAlgorithm,
+            Span<byte> destination)
+        {
+            Pbkdf2DeriveBytesCore(password, salt, iterations, hashAlgorithm, destination);
+        }
+
+        public static byte[] Pbkdf2DeriveBytes(
+            string password,
+            byte[] salt,
+            int iterations,
+            int length,
+            HashAlgorithmName hashAlgorithm)
+        {
+            if (password is null)
+                throw new ArgumentNullException(nameof(password));
+            if (salt is null)
+                throw new ArgumentNullException(nameof(salt));
+
+            return Pbkdf2DeriveBytes(password.AsSpan(), new ReadOnlySpan<byte>(salt), iterations, length, hashAlgorithm);
+        }
+
+        public static byte[] Pbkdf2DeriveBytes(
+            ReadOnlySpan<char> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            int length,
+            HashAlgorithmName hashAlgorithm)
+        {
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            byte[] result = new byte[length];
+            Pbkdf2DeriveBytes(password, salt, iterations, hashAlgorithm, result);
+            return result;
+        }
+
+        public static void Pbkdf2DeriveBytes(
+            ReadOnlySpan<char> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            HashAlgorithmName hashAlgorithm,
+            Span<byte> destination)
+        {
+            byte[] passwordBytes = CryptoPool.Rent(Encoding.UTF8.GetMaxByteCount(password.Length));
+            int passwordBytesWritten = Encoding.UTF8.GetBytes(password, passwordBytes);
+
+            Pbkdf2DeriveBytesCore(passwordBytes, salt, iterations, hashAlgorithm, destination);
+            CryptoPool.Return(passwordBytes, clearSize: passwordBytesWritten);
+        }
+
+        private static void Pbkdf2DeriveBytesCore(
+            ReadOnlySpan<byte> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            HashAlgorithmName hashAlgorithm,
+            Span<byte> destination)
+        {
+            if (salt.Length < MinimumSaltSize)
```

Comment:
Maybe we don't want a salt limit? IIRC, one of the blockers from DataProtection moving off their own implementation was they permit any size salt. Perhaps we should do the same, here.

Reply:
I think this is probably the right way to go, though it is a discrepancy from the instance-based implementation. I went ahead and made the change, but, my current thought is: there should be no minimum. But, if we want to get in the business of minimums, eight was probably too-low of a recommendation, anyway.